xds/internal/client/client_xds.go

```diff
@@ -134,19 +134,13 @@ func processServerSideListener(lis *v3listenerpb.Listener) (*ListenerUpdate, err
 		return nil, fmt.Errorf("xds: socket_address port does not match the one in name. Got %q, want %q", p, port)
 	}
 
-	// Make sure that the listener resource contains a single filter chain with
-	// the application_protocols field set to "managed-mtls", and the
-	// "transport_socket" field containing the appropriate security
-	// configuration.
+	// Make sure the listener resource contains a single filter chain. We do not
+	// support multiple filter chains and picking the best match from the list.
 	fcs := lis.GetFilterChains()
 	if n := len(fcs); n != 1 {
 		return nil, fmt.Errorf("xds: filter chains count in LDS response does not match expected. Got %d, want 1", n)
 	}
 	fc := fcs[0]
-	aps := fc.GetFilterChainMatch().GetApplicationProtocols()
```

Comment:
Just a thought: would it make sense to leave a comment here?
Thinking probably no? Since there are plenty fields that we don't check. And when a decision is made, we will come back and add the required check anyway.

Reply:
A bug is open where the discussion is happening. So, once the decision is made, I think all three languages will come back and implement the decision. So, I'm ok not adding a comment here. Thanks.